src/main/groovy/com/brightsparklabs/gradle/baseline/BaselinePlugin.groovy

```diff
@@ -351,45 +358,139 @@ public class BaselinePlugin implements Plugin<Project> {
             group = "brightSPARK Labs - Baseline"
-            description = "Upload files to an S3 bucket. Configure via the `bslBaseline` configuration block."
+            description = "Upload files to an S3 bucket. Configure via the `bslBaseline`" +
+                    " configuration block."
 
             doLast {
                 final S3ClientBuilder s3Builder = S3Client.builder()
+
                 // By default, the AWS SDK will attempt to pull the region from the system.
                 // If configured, we allow for an optional override.
                 if (!Strings.isNullOrEmpty(region)) {
                     s3Builder.region(Region.of(region))
                 }
-                final S3Client s3 = s3Builder.build()
+
+                // By default, the AWS SDK will use the "default" profile from the system.
+                // If configured, we allow for an optional override.
+                if (!Strings.isNullOrEmpty(profile)) {
+                    final ClientOverrideConfiguration.Builder s3OverrideConfigBuilder =
+                            ClientOverrideConfiguration.builder()
+                    s3OverrideConfigBuilder.defaultProfileName(profile)
+                    final s3OverrideConfig = s3OverrideConfigBuilder.build()
+                    s3Builder.overrideConfiguration(s3OverrideConfig)
+                }
+
+                // By default, the AWS SDK will use the default S3 endpoint for the given region.
+                // If configured, we allow for an optional override.
+                if (!Strings.isNullOrEmpty(endpointOverride)) {
+                    URI endpointOverrideURI
+                    try {
+                        endpointOverrideURI = new URI(endpointOverride)
+                    } catch (URISyntaxException e) {
+                        logger.error(
+                                "A URISyntaxException occurred while parsing the" +
+                                " `deploy.s3.endpointOverride` configuration option. Ensure that" +
+                                " the given value is a valid URI."
+                                )
+                        throw e
+                    }
+                    s3Builder.endpointOverride(endpointOverrideURI)
+                }
+
+                S3Client s3Temp
+                try {
+                    s3Temp = s3Builder.build()
+                } catch (SdkClientException e) {
+                    logger.error(
+                            "An SdkClientException occurred. This may have been caused by an" +
+                            " incorrect `deploy.s3.profile` configuration. If the configuration" +
+                            " is correct, ensure that the profile exists in the system, and" +
+                            " \"Default region name\" is set to a non-empty string."
+                            )
+                    throw e
+                }
+                final S3Client s3 = s3Temp
```

Comment:
There's no point using two variables.
Just rename the top one to `s3` and get rid of this one.
You'll throw if it fails, so there's no difference between `s3Temp` and `s3`.
If the `final` is what is causing you issue, then drop it, but it shouldn't be a problem.